lib/bundler/definition.rb

```diff
@@ -200,8 +200,10 @@ def resolve
       @resolve ||= begin
         last_resolve = converge_locked_specs
         if Bundler.settings[:frozen] || (!@unlocking && nothing_changed?)
+          Bundler.ui.debug("Resolved from lockfile")
           last_resolve
         else
+          Bundler.ui.debug("Resolving again")
```

Comment:
Note, this is the exact place I put `STDERR.puts` calls based on @segiddins guidance that if we have a lockfile, we shouldn't be entering the "Resolving again" condition.  I added this logging for the test specifically, but I also think it's useful if people report  "bundle exec XYZ is slow".

This is what it looks like:

```
$ DEBUG=1 bundle check
Resolved from lockfile
The Gemfile's dependencies are satisfied
```

Now, I added a gem to my Gemfile...

```
$ DEBUG=1 bundle check
Resolving again
Bundler can't satisfy your Gemfile's dependencies.
```